src/KDMqtt/mqtt.h

```diff
@@ -0,0 +1,250 @@
+/*
+  This file is part of KDUtils.
+
+  SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+  Author: Marco Thaller <marco.thaller@kdab.com>
+
+  SPDX-License-Identifier: MIT
+
+  Contact KDAB at <info@kdab.com> for commercial licensing options.
+*/
+#pragma once
+
+#include <KDFoundation/file_descriptor_notifier.h>
+#include <KDFoundation/timer.h>
+#include <KDMqtt/kdmqtt_global.h>
+#include <KDMqtt/mosquitto_wrapper.h>
+#include <KDUtils/file.h>
+#include <KDUtils/url.h>
+#include <unordered_map>
+
+using namespace KDFoundation;
+using namespace KDUtils;
+
+namespace KDMqtt {
+
+constexpr int c_defaultPort = 1883;
+constexpr int c_defaultKeepAliveSeconds = 60;
+
+/*
+ * Class: IMqttLib
+ *
+ * This is an abstract class specifing a generic
+ * interface exposed to application / business logic
+ * to access MQTT library implementations.
+ */
+class KDMQTT_API IMqttLib
+{
+protected:
+    IMqttLib() = default;
+    virtual ~IMqttLib() { }
+
+public:
+    IMqttLib(const IMqttLib &) = delete;
+    IMqttLib &operator=(const IMqttLib &) = delete;
+
+    virtual int init() = 0;
+    virtual int cleanup() = 0;
+
+    [[nodiscard]] virtual bool isInitialized() const = 0;
+    virtual bool isValidTopicNameForSubscription(const std::string &topic) = 0;
+};
+
+/*
+ * Class: MqttLib
+ *
+ * This class exposes mosquitto library functions to
+ * the application / business logic.
+ */
+class KDMQTT_API MqttLib : public IMqttLib
+{
+    friend class MqttClient;
+    friend class MqttUnitTestHarness;
+
+private:
+    MqttLib();
+    ~MqttLib() = default;
```

Comment:
I think that this class should call `cleanup()` on destruction. So that the user doesn't have to think about it (case in point: cleanup is not done in the example :) ). 

According to mosquitto docs, mosquitto_lib_cleanup cannot fail so we can disregard its return value.